ros_buildfarm/templates/snippet/reconfigure_jobs.groovy.em

```diff
@@ -102,14 +84,27 @@ def format_xml(doc) {
 
 // reconfigure jobs
 println '# BEGIN SECTION: Groovy script - reconfigure jobs'
-println 'Reconfiguring ' + job_configs.size() + ' jobs...'
+println 'Reconfiguring @(expected_num_jobs) jobs...'
 created = 0
 updated = 0
 skipped = 0
-for (item in job_configs) {
+
+config_dir = build.getWorkspace().toString() + '/reconfigure_jobs/configs'
+
+def jobs = []
+def dir = new File(config_dir)
+dir.eachFileRecurse (FileType.FILES) { file ->
+    jobs << file
+}
+
+if (jobs.size() != @(expected_num_jobs)) {
+    println "WARNING: Found different number of job configs than expected!! " + jobs.size() + " is not @(expected_num_jobs) as expected."
```

Comment:
Should this really only be a warning? I would think if this happens failing the job and sending a notification email would be more appropriate.